platform/pulp/platform/tasks/task_system.py

```diff
@@ -1,4 +1,548 @@
+
+        class MyEinfo(object):
+            traceback = None
+
+        new_task.on_failure(runtime_exception, task_id, (), {}, MyEinfo)
+
+    ReservedResource.objects.filter(task__id=task_id).delete()
+
+
+class TaskResult(object):
+    """
+    The TaskResult object is used for returning errors and spawned tasks that do not affect the
+    primary status of the task.
+
+    Errors that don't affect the current task status might be related to secondary actions
+    where the primary action of the async-task was successful
+
+    Spawned tasks are items such as the individual tasks for updating the bindings on
+    each consumer when a repo distributor is updated.
+    """
+
+    def __init__(self, result=None, error=None, spawned_tasks=None):
+        """
+        :param result: The return value from the task
+        :type result: dict
+        :param error: The PulpException for the error & sub-errors that occurred
+        :type error: pulp.server.exception.PulpException
+        :param spawned_tasks: A list of task status objects for tasks that were created by this
+                              task and are tracked through the pulp database.
+                              Alternately an AsyncResult, or the task_id of the task created.
+        :type spawned_tasks: list of TaskStatus, AsyncResult, or str objects
+        """
+        self.return_value = result
+        self.error = error
+        self.spawned_tasks = []
+        if spawned_tasks:
+            for spawned_task in spawned_tasks:
+                if isinstance(spawned_task, dict):
+                    self.spawned_tasks.append({'task_id': spawned_task.get('task_id')})
+                elif isinstance(spawned_task, AsyncResult):
+                    self.spawned_tasks.append({'task_id': spawned_task.id})
+                elif isinstance(spawned_task, TaskStatus):
+                    self.spawned_tasks.append({'task_id': spawned_task.task_id})
+                else:  # This should be a string
+                    self.spawned_tasks.append({'task_id': spawned_task})
+
+    @classmethod
+    def from_async_result(cls, async_result):
+        """
+        Create a TaskResult object from a celery async_result type
+
+        :param async_result: The result object to use as a base
+        :type async_result: celery.result.AsyncResult
+        :returns: a TaskResult containing the async task in it's spawned_tasks list
+        :rtype: TaskResult
+        """
+        return cls(spawned_tasks=[{'task_id': async_result.id}])
+
+    @classmethod
+    def from_task_status_dict(cls, task_status):
+        """
+        Create a TaskResult object from a celery async_result type
+
+        :param task_status: The dictionary representation of a TaskStatus
+        :type task_status: dict
+        :returns: a TaskResult containing the task in it's spawned_tasks lsit
+        :rtype: TaskResult
+        """
+        return cls(spawned_tasks=[{'task_id': task_status.task_id}])
```

Comment:
The next revision will have a lot of this code deleted. The `TaskStatus` object is also going to be made available through the `pulp.plugin` API.